daemon/emer-network-send-provider.c

```diff
@@ -191,15 +191,19 @@ reset_network_send_data (EmerNetworkSendProvider *self)
                   error->message);
       g_error_free (error);
     }
+
+  priv->send_number = 0;
+  priv->data_cached = TRUE;
 }
 
-static gboolean
+static void
 read_network_send_data (EmerNetworkSendProvider *self)
 {
   EmerNetworkSendProviderPrivate *priv =
     emer_network_send_provider_get_instance_private (self);
+
   if (priv->data_cached)
```

Comment:
Thanks; had trouble finding that. Personally, I tend to find initializing values explicitly easier to read as evidenced by https://github.com/endlessm/eos-event-recorder-daemon/blob/master/daemon/emer-daemon.c#L1409-L1410 and other places in the metrics system. We work with a lot of different programming languages, and I have a hard time keeping track of which ones automatically initialize variables under which circumstances. It's also more obviously wrong if a variable is explicitly initialized to 0 but should have been initialized to some other value. I don't have very strong sentiments about it now that we are very certain that it does the right thing either way.